pkg/target/target_test.go

```diff
@@ -94,13 +106,47 @@ func TestProcessLabelValues(t *testing.T) {
 	if !ok {
 		t.Fatal("unable to assert third element of customStruct to map[string]interface{}")
 	}
-
 	thirdElemVal, ok := thirdElement["element2"]
 	if !ok {
 		t.Fatal("unable to find key element2")
 	}
-
 	if thirdElemVal.(string) != "local" {
 		t.Fatal("label replacement not performed in third element")
 	}
+
+	templateName, ok := bundle.Helm.Values.Data["templateName"]
+	if !ok {
+		t.Fatal("key templateName not found")
+	}
+	if templateName != "kubernetes.io/cluster/local" {
+		t.Fatal("unable to assert correct template")
+	}
+
+	templateNameDash, ok := bundle.Helm.Values.Data["template-name"]
+	if !ok {
+		t.Fatal("key template-name not found")
+	}
+	if templateNameDash != "kubernetes.io/cluster/c-test" {
+		t.Fatal("unable to assert correct template-name")
+	}
+
+	templateLogic, ok := bundle.Helm.Values.Data["templateLogic"]
+	if !ok {
+		t.Fatal("key templateLogic not found")
+	}
+	if templateLogic != "Non Prod" {
+		t.Fatal("unable to assert correct templateLogic")
+	}
+
+	bundle.Helm.Values.Data["error"] = "{{ .missingValue }}"
+	err = processValues(bundle.Helm.Values.Data, clusterLabels, clusterAnnotations)
+	if err == nil {
+		t.Fatalf("failed to error on {{ .missingValue }}")
+	}
+
+	bundle.Helm.Values.Data["error"] = "{{ .missingValue-panic }}"
+	err = processValues(bundle.Helm.Values.Data, clusterLabels, clusterAnnotations)
+	if err == nil {
+		t.Fatalf("failed to error on {{ .missingValue-panic }}")
+	}
```

Comment:
How are these different tests?

Reply:
One is designed to raise a panic from the go templating.  I ran into a panic at one point in time so I wrote a test for it.  If I recall correctly the `-` in the variable name actually causes the issue.